### PR TITLE
[docs] Add a doc section on how to override the start of the week with each adapter

### LINKED ISSUE #9785: add weekFormatter just like dayOfWeekFormatter
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

requesting a prop like dayOfWeekFormatter for the week numbers

### Examples 🌈

https://github.com/search?q=repo%3AHacker0x01%2Freact-datepicker%20formatweeknumber&type=code

### Motivation 🔦

I am trying to change the week numbers but not able to. using a different way of calculating week numbers (for organization) that differs from the current standard formats

### Order ID 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #9785

I initially added demos but since the locale configuration is global, it impacts all the other demos :grimacing:

### CODE REVIEW

File: docs/data/date-pickers/adapters-locale/adapters-locale.md [hunk 1]
```diff
@@ -80,31 +80,6 @@ import 'moment/locale/de';
 
 {{"demo": "LocalizationMoment.js"}}
 
-:::warning
```
Comment (flaviendelangle) [inline docs/data/date-pickers/adapters-locale/adapters-locale.md:83]: We are no longer using any `moment` method that are not localized.

**Review by LukasTy — COMMENTED** [review]
Nice addition to the doc. 👍 

I've just now checked that we might want to reconsider the release of this Luxon localized feature, because it's not going to work on Firefox. 🙈 🤯 
https://moment.github.io/luxon/#/intl?id=locale-based-weeks
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo#browser_compatibility
WDYT? On one hand, it's still an improvement for most, but on the other hand, it could be an annoying issue for others. 🤔 
Maybe we should at least add a callout informing that currently Firefox does not support the localized Luxon behavior? 🤔 
P.S. I was unable to locate an active firefox issue tracking the progress of this feature delivery. 🤷

**Comment by flaviendelangle** [comment]
> I've just now checked that we might want to reconsider the release of this Luxon localized feature, because it's not going to work on Firefox. 🙈 🤯


:sob: 
I can look how it behaves on Firefox, but I guess we can make it opt-in and wait for Firefox to support it

**Comment by LukasTy** [comment]
> I can look how it behaves on Firefox

Well, it simply doesn't react to the language/locale. Having English(en-us) set still renders the calendar the ISO way (starts on Monday).

**Comment by flaviendelangle** [comment]
How do you think we should handle this feature then?

- Keep if opt-out
- Make it opt-in
- Totally revert the changes

**Comment by flaviendelangle** [comment]
I'm fine with this approach.

Would you do something like we proposed originally in this API ?

````tsx
<LocalizationProvider adapter={AdapterLuxon.withLocaleWeeks()}>
</LocalizationProvider>
```

Or do you have another API in mind?

File: docs/data/date-pickers/adapters-locale/adapters-locale.md [hunk 2]
```diff
@@ -270,3 +245,95 @@ This prop is available on all components that render a day calendar, including t
...
+  // Sunday = 7, Monday = 1.
+  firstDay: 1,
+  // Makes sure we don't lose the other information from `defaultWeekSettings`
+  minimalDays: Info.getMinimumDaysInFirstWeek(),
+  weekend: Info.getWeekendWeekdays(),
+};
+```
+
+:::warning
+The browser API used by Luxon to determine the start of the week in the current locale is not yet supported by Firefox.
```
Comment (LukasTy) [inline docs/data/date-pickers/adapters-locale/adapters-locale.md:308]:
```suggestion
The [browser API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale/getWeekInfo) used by Luxon to determine the start of the week in the current locale is not yet supported by Firefox.
```

**Review by LukasTy — APPROVED** [review]
LGTM! 💯 🚀

**Pull request merged**